SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::AsyncGenerateDevAuthToken()
+{
+	bool bExpected = false;
+	if (bIsGenerating.CompareExchange(bExpected, true))
```

Comment:
```suggestion
	if (!bIsGenerating.CompareExchange(bExpected, true))
	{
		UE_LOG(LogSpatialGDKDevAuthTokenGenerator, Display, TEXT("Developer Authentication Token requested but a previous request is still pending. New request for generation ignored."));
		return;
	}
	DevAuthToken.Empty();
```
It is better to return directly here so we don't need to write the normal logic into brackets. That will make code looks clearer.

Reply:
I prefer if() else .. it makes move method refactor more easy.

Reply:
I prefer raymon's suggestion. It makes the code easier to read and allows us to early out